ycmd/tests/testdata/general_fallback/lang_c.c

```diff
@@ -0,0 +1,85 @@
+// Identifier completion makes sense for duck-typing.
+//
+// The purpose of this test is to both demonstrate the use-case and test the
+// functionality. Completions of "A.a_parameter" using the identifier
+// completer feels natural, whereas offering no suggestions feels broken
+
+typedef struct {
+  int a_parameter;
+  int another_parameter;
+} TTest;
+
+typedef struct {
+  int another_int;
+  int and_a_final_int;
+} TTest_Which_Is_Not_TTest;
+
+static void do_something( int );
+static void do_another_thing( int );
+
+// TESTCASE1: use of . on macro parameter. Macros used this way are compile-time
+// duck-typing. Note this in this particular instance, libclang actually
+// offers semantic completions if there is only call to this macro
+// (e.g. DO_SOMETHING_TO( a_test ) - clever little shrew), so we don't call it
+
+/*       1         2         3         4
+1234567890123456789012345678901234567890123456789 */
```

Comment:
Yeah, it's a ruler. I find reading the tests and the associated code quite a painfully slow and cumbersome when you have to read 2 numbers and then navigate to that exact spot in the file.

You can get vim to show column numbers, of course, but you still have to pretty much put the cursor there. The use case for the ruler is a vertical split where i'm writing (or reading) the test in one split and i can just visually see (via set nu and this ruler) the exact spot where the test needs to trigger (or is already triggering).

I'm happy to remove them if you feel strongly about it. Of course removing them changes all the offsets which is a bit tedious, but hey ho :)